core/src/HIP/Kokkos_HIP_Team.hpp

```diff
@@ -243,8 +247,11 @@ class HIPTeamMember {
 
     base_data[threadIdx.y + 1] = value;
 
-    Impl::hip_intra_block_reduce_scan<true, Impl::HIPJoinFunctor<Type>, void>(
-        Impl::HIPJoinFunctor<Type>(), base_data + 1);
+    Impl::HIPJoinFunctor<Type> hip_join_functor;
+    typename Kokkos::Impl::FunctorAnalysis<
+        FunctorPatternInterface::REDUCE, TeamPolicy<Experimental::HIP>,
+        Impl::HIPJoinFunctor<Type>>::Reducer reducer(&hip_join_functor);
+    Impl::hip_intra_block_reduce_scan<true>(reducer, base_data + 1);
```

Comment:
... and here allow for all the deletions in [core/src/HIP/Kokkos_HIP_Shuffle_Reduce.hpp](https://github.com/kokkos/kokkos/pull/4836/files#diff-aae11646146b93dffa63de8dc2675e97d26fcace62a0f7a4b93250837d01f53d).